server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java

```diff
@@ -291,16 +291,40 @@ public int hashCode() {
   private final ServerContext context;
   private FSDataOutputStream logFile;
   private DataOutputStream encryptingLogFile = null;
-  private LogEntry logEntry;
+  private final LogEntry logEntry;
   private Thread syncThread;
 
   private AtomicLong syncCounter;
   private AtomicLong flushCounter;
   private final long slowFlushMillis;
   private long writes = 0;
 
-  public DfsLogger(ServerContext context, AtomicLong syncCounter, AtomicLong flushCounter) {
-    this(context, null);
+  public static DfsLogger fromCounters(ServerContext context, AtomicLong syncCounter,
```

Comment:
I'm not sure of the best name for this method but maybe something like `fromNewLogEntry` might be a good fit too. I am also fine with leaving it as is so unless someone else has an opinion I think its fine.

Reply:
I pushed a different name in my code review commit, in collab with @ArbaazKhan1